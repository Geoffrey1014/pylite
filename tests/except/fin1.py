def main():
    try:
        print 'main start'
        0/0
        print 'main end'
    finally:
        print 'finally' 

print 'global start'
main()
print 'global end'
