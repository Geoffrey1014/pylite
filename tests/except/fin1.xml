lambda
  name __init__
  param_count 0
  local_count 0
  label_count 0
  insn_vector
    load_const c0
    store_global c1
    load_const c2
    print
    load_global c1
    call 0
    pop
    load_const c3
    print
    load_const c4
    return
  const_vector
    lambda
      name main
      param_count 0
      local_count 1
      label_count 4
      insn_vector
        label L0
        load_const c0
        print
        load_const c1
        load_const c1
        div
        pop
        load_const c2
        print
        label L1
        jmp L3
        label L2
        store_local -1
        load_const c3
        print
        finally_exit -1
        label L3
        finally_call L2
        load_const c4
        return
      const_vector
        "main start"
        0.000000
        "main end"
        "finally"
        None
      guard_vector
        guard
          start L0
          end L1
          handler L2
      line_number_map 3 3 3 4 4 4 4 5 5 3 3 3 3 7 7 3 3 3 1 1
    main
    "global start"
    "global end"
    None
  guard_vector
  line_number_map 1 1 9 9 10 10 10 11 11 1 1
