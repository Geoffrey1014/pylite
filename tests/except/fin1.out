global start
main start
finally
ZeroDivisionError:
  At fin1.py, line 4, in main()
  At fin1.py, line 10, in __init__()
